### ISSUE TITLE
[BUG] cudf Index's attributes like is_monotonic do not have values

### ISSUE THREAD

**Issue by qinxuye** [rapidsai/cudf#2150]
**Describe the bug**

For our project which aims at distributed dataframe computation, index's attributes like `is_monotonic` is used to optimize the execution, if we know that the index is monotonic increasing in advance, we could avoid shuffle for some index operations. But cudf's Index including `RangeIndex` etc do not have value or do not exist.

The attributes including:

* is_monotonic
* is_monotonic_increasing
* is_monotonic_decreasing
* is_unique

**Steps/Code to reproduce bug**

```
In [1]: import cudf 

In [9]: index = cudf.dataframe.index.RangeIndex(3)                                                                                                                                                                 

In [10]: index.is_monotonic 

In [13]: index.is_unique                                                                                                                                                                                           
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-13-ccf673644855> in <module>
----> 1 index.is_unique

AttributeError: 'RangeIndex' object has no attribute 'is_unique'                                                                                                                                                                                       

In [14]: index.is_monotonic_decreasing                                                                                                                                                                             

In [15]: index.is_monotonic_increasing                                                                                                                                                                             
```

**Expected behavior**

The attributes like `is_monotonic` have value.

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: Conda

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

```
**git***
Not inside a git repository

***OS Information***
Alibaba Group Enterprise Linux Server release 7.2 (Paladin)
NAME="Alibaba Group Enterprise Linux Server"
VERSION="7.2 (Paladin)"
ID="alios"
ID_LIKE="fedora"
VERSION_ID="7.2"
PRETTY_NAME="Alibaba Group Enterprise Linux Server 7.2 (Paladin)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:alibaba:enterprise_linux:7.2:GA:server"
HOME_URL="https://www.aliyun.com/"
BUG_REPORT_URL="http://bbs.aliyun.com/"

ALIBABA_BUGZILLA_PRODUCT="Alibaba Group Enterprise Linux 7"
ALIBABA_BUGZILLA_PRODUCT_VERSION=7.2
ALIBABA_SUPPORT_PRODUCT="Alibaba Group Enterprise Linux"
ALIBABA_SUPPORT_PRODUCT_VERSION=7.2
Alibaba Group Enterprise Linux Server release 7.2 (Paladin)
Alibaba Group Enterprise Linux Server release 7.2 (Paladin)
Linux tb1a01277.sqa.tbc 3.10.0-327.ali2012.alios7.x86_64 #1 SMP Mon Oct 9 14:09:14 CST 2017 x86_64 x86_64 x86_64 GNU/Linux

***GPU Information***
Sat Jun 29 13:20:23 2019       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.67       Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:1B:00.0 Off |                    0 |
| N/A   31C    P0    39W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla V100-SXM2...  Off  | 00000000:1C:00.0 Off |                    0 |
| N/A   32C    P0    41W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla V100-SXM2...  Off  | 00000000:3D:00.0 Off |                    0 |
| N/A   33C    P0    40W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla V100-SXM2...  Off  | 00000000:3E:00.0 Off |                    0 |
| N/A   31C    P0    38W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   4  Tesla V100-SXM2...  Off  | 00000000:B1:00.0 Off |                    0 |
| N/A   32C    P0    38W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   5  Tesla V100-SXM2...  Off  | 00000000:B2:00.0 Off |                    0 |
| N/A   33C    P0    38W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   6  Tesla V100-SXM2...  Off  | 00000000:DB:00.0 Off |                    0 |
| N/A   32C    P0    38W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   7  Tesla V100-SXM2...  Off  | 00000000:DC:00.0 Off |                    0 |
| N/A   31C    P0    39W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

***CPU***
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                96
On-line CPU(s) list:   0-95
Thread(s) per core:    2
Core(s) per socket:    24
Socket(s):             2
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Platinum 8163 CPU @ 2.50GHz
Stepping:              4
CPU MHz:               2500.000
CPU max MHz:           3100.0000
CPU min MHz:           1000.0000
BogoMIPS:              4998.90
Virtualization:        VT-x
Hypervisor vendor:     vertical
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              33792K
NUMA node0 CPU(s):     0-95
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch ida arat epb pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx avx512f avx512dq rdseed adx smap clflushopt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 cqm_llc cqm_occup_llc

***CMake***
which: no cmake in (/sbin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr/X11R6/bin:/home/tops/bin:/home/xuye.qin/.local/bin:/home/xuye.qin/bin)

***g++***
/usr/bin/g++
g++ (GCC) 4.8.5 20150623 (Red Hat 4.8.5-28)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


***nvcc***
which: no nvcc in (/sbin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr/X11R6/bin:/home/tops/bin:/home/xuye.qin/.local/bin:/home/xuye.qin/bin)

***Python***
/usr/bin/python
Python 2.7.5

***Environment Variables***
PATH                            : /sbin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr/X11R6/bin:/home/tops/bin:/home/xuye.qin/.local/bin:/home/xuye.qin/bin
LD_LIBRARY_PATH                 : :/usr/local/cuda/lib64
NUMBAPRO_NVVM                   : 
NUMBAPRO_LIBDEVICE              : 
CONDA_PREFIX                    : 
PYTHON_PATH                     : 

conda not found
pip not found
```

**Comment by beckernick** [comment]
This may be relevant for someone looking into this issue: https://github.com/pandas-dev/pandas/blob/96dd2a95c21321608645d8195aebb48519618a54/pandas/_libs/algos.pyx#L704

Thanks for filing this @qinxuye

**Comment by kkraus14** [comment]
@qinxuye We added `None` for these attributes just to appease something upstream which checked if the attribute exists before checking the values for optimization. If we had these always return False for now would that unblock you until we have a proper implementation to return a usable value?

**Comment by qinxuye** [comment]
> @qinxuye We added `None` for these attributes just to appease something upstream which checked if the attribute exists before checking the values for optimization. If we had these always return False for now would that unblock you until we have a proper implementation to return a usable value?

Yeah, I can accept False for now until they have reasonable values.

**Comment by kkraus14** [comment]
@rjzamora do you want to try to tackle this? Quick PR to set them to False and then a bigger PR implementing functions to get the correct value?

**Comment by rjzamora** [comment]
No problem @kkraus14 - I'll take a look

**Comment by kkraus14** [comment]
Fixed by #2162
